### ISSUE TITLE
Issues using cryptomatte in htoa

### ISSUE THREAD

**Issue by javiertoledovis** [anderslanglands/alShaders2#14]
Hi there.

We are testing cryptomatte in arnold for houdini (htoa 2.1.4, arnold
5.0.1.4, houdini 16.0.736). Everything works fine except for a few things.

**1. Material names in nuke:** When we read cryptomattes in nuke we can
select objects by name, such as /obj/sphere1, but with shaders it's not
working. Instead of /shop/blueMetal this appears: <-2.99242719543e+16>

**2. User defined cryptomattes:** We use user data to identify different
groups of polygons or points (i tried both polygon and point data) inside
of an object. When I define User cryptomatte 0 with an AOV name and the
user data name the aov renders black. I think i followed the git wiki but I
cant make it work.

**3. Not clean crypto_asset_offset:** When i use an integer offset in an
object to select a subset of polygons it works but the colors are not
clean, when I select an object  parts of the others are selected as well.

Thank you for your time!

Cheers

**Comment by jonahfriedman** [comment]
Hi @javiertoledovis, thanks for reporting. 

Could you double check which version of AlShaders you're using? #1 in particular looks like something we already fixed. If you are on latest (2.0.0b2), reproductions for all of these would helpful (ideally in the form of ass files). 

1. (Material names) This sounds like a problem that was fixed. 
2. (Custom Cryptomatte) Is your user data of type string? That's the only supported type for custom cryptomattes at the moment. 
3. (Non-clean offsets) I'm not sure what could cause this- could you send an ass file that renders this way and an EXR?

**Comment by javiertoledovis** [comment]
Hi @jonahfriedman 

Thanks for answering. 
I noticed that i was using the b1 version, I just updated to b2.

1 (Material names)- Solved!
2 (Custom cryptomatte)- I just tried setting the user data in polygon level instead of point level and it's working. It didn´t work in b1 but it's working in b2. Great!
3 (Non-clean offsets)- Same solution as previous point.

Everything is working as expected. Thanks so much!
